### PR TITLE
Change HADOOP_TARGET_VERSION tokens to hadoop, cdh, hdp, mpr

### PR DESCRIPTION
**Pull request by lavjain**

### CODE REVIEW

File: configure.in [hunk 1]
```diff
@@ -230,6 +230,13 @@ PGAC_ARG_BOOL(enable, pxf, yes,
              [do not build pxf])
 AC_SUBST(enable_pxf)
 
+#
+# gphdfs
+#
+PGAC_ARG_BOOL(enable, gphdfs, yes,
+             [do not build gphdfs])
+AC_SUBST(enable_gphdfs)
```
Comment (danielgustafsson) [inline configure.in]: This doesn't check for any pre-requisites or dependencies, surely there are requirements we should ensure are in place?

**Comment by lisakowen** [comment]
@lavjain - if we are removing some of the gp_hadoop_target_version values, should we update the default value of this GUC since it is currently set to one of the values removed in this PR?

backend/utils/misc/guc_gp.c:

{
                {"gp_hadoop_target_version", PGC_USERSET, CUSTOM_OPTIONS,
                        gettext_noop("The distro/version of Hadoop that external table is connecting to."),
                        gettext_noop("See release notes or gppkg readme for details."),
                        GUC_GPDB_ADDOPT
                },
                &gp_hadoop_target_version,
                "gphd-1.1", NULL, NULL
        },

File: configure.in (hunk 1 above)
Comment (lavjain) [reply configure.in]: @danielgustafsson The change was modeled after pxf and gpcloud (so all extensions are consistent). However, the default value will be reverted to no for enable-gphdfs. I believe the only dependency is "curl", but did not see it in the previous code.

**Comment by lavjain** [comment]
@lisakowen Thanks a lot for catching the omission.

File: gpAux/extensions/gphdfs/regression/run_gphdfs_regression.sh [hunk 2]
```diff
@@ -3,8 +3,8 @@
 set -euo pipefail
 
 # set variables if they are unset or null
-export HADOOP_HOME=${HADOOP_HOME:-/usr/hdp/2.3.2.0-2950}
-export GP_HADOOP_TARGET_VERSION=${GP_HADOOP_TARGET_VERSION:-cdh4.1}
+export HADOOP_HOME=${HADOOP_HOME:-/usr/hdp/current/hadoop-client}
+export GP_HADOOP_TARGET_VERSION=${GP_HADOOP_TARGET_VERSION:-hadoop}
```
Comment (denalex) [inline gpAux/extensions/gphdfs/regression/run_gphdfs_regression.sh]: are we using Hadoop target but still expect HDP-based installation ? Would Hadoop target mean generic Hadoop distro that would not be installed in /usr/hdp/current ?

Comment (lavjain) [reply gpAux/extensions/gphdfs/regression/run_gphdfs_regression.sh]: I agree. My attempt was an improvement over the previous logic :-)

**Pull request merged**